python/ray/dataframe/index_metadata.py

```diff
@@ -203,15 +203,15 @@ def reset_partition_coords(self, partitions=None):
             # updated as well.
             try:
                 self._coord_df.loc[partition_mask,
-                                   'index_within_partition'] = [
-                    p for p in range(sum(partition_mask))]
+                                   'index_within_partition'] = np.arange(
```

Comment:
The previous method will prevent consecutive del calls:
```
del df['col1']
del df['col2'] # Fail because the col_metadata contains float
```
It seems setting with loc requires extra caution here. That's why I added type casting.